plugins/modules/idrac_system_erase.py

```diff
@@ -0,0 +1,569 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+#
+# Dell OpenManage Ansible Modules
+# Version 9.7.0
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+#
+
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: idrac_system_erase
+short_description: Erase system and storage components of the server
+version_added: "9.7.0"
+description:
+  - This module allows erasing System components such as iDRAC, BIOS, DIAG,
+    etc., and Storage components such as PERC NV cache, Non-volatile memory,
+    Cryptographic erase PD, etc. of the server.
+extends_documentation_fragment:
+  - dellemc.openmanage.idrac_x_auth_options
+options:
+  component:
+    description:
+      - System and storage components to erase.
+      - Below are the supported components.
+        AllApps
+        BIOS
+        CryptographicErasePD
+        DIAG
+        DPU
+        DrvPack
+        IDRAC
+        LCData
+        NonVolatileMemory
+        OverwritePD
+        PERCNVCache
+        ReinstallFW
+        vFlash
+    type: list
+    elements: str
+    required: true
+  power_on:
+    description:
+      - This allows to power on the server after erase operation. This is applicable only when I(job_wait) is C(true).
```

Comment:
This parameter allows you to power on the server after the delete operation is completed. This is applicable when I(job_wait) is C(true).